### PR TITLE
Calculate fingerprints of TLS certificates

### LINKED ISSUE #8180: [Packetbeat] Add SHA1 Fingerprints of TLS Certificates
As a user I'd like to be able to search for certificates based on SHA1 fingerprints. For example if I want to query for certificates matching a SSL blacklist like https://sslbl.abuse.ch/ then I need to have a fingerprint in Elasticsearch.

Packetbeat should probably also support SHA256 hashing as well. Some apps are moving to using that.

- CSV Blacklist - https://sslbl.abuse.ch/downloads/ssl_extended.csv

### PR DESCRIPTION
**Pull request by adriansr**
This patch adds support to calculate fingerprints to the TLS parser in packetbeat. The feature is controlled by a new configuration option:
```
fingerprints: [sha1, sha256]
```

Valid values are `sha1`, `sha256` and `md5`.

Closes #8180

### CODE REVIEW

File: packetbeat/protos/tls/config.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ type tlsConfig struct {
 	config.ProtocolCommon  `config:",inline"`
 	SendCertificates       bool     `config:"send_certificates"`
 	IncludeRawCertificates bool     `config:"include_raw_certificates"`
-	Fingerprints           []string `config:fingerprints`
+	Fingerprints           []string `config:"fingerprints"`
 }
 
 var (
```
Comment (andrewkroh) [inline packetbeat/protos/tls/config.go]: Shouldn't `defaultConfig` contain `fingerprints: []string{"sha1"}`?

Comment (adriansr) [reply packetbeat/protos/tls/config.go]: I thought it was better to have `fingerprints: [sha1]` added by default to the config file, as it is now, and treat it as an empty list if removed. But that may not fit well with our configuration style, I see the rest of protocols don't have any other configuration option set other than ports.

**Comment by andrewkroh** [comment]
jenkins test this

**Comment by adriansr** [comment]
Test failure in TLS, will investigate

**Pull request merged**